micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jetty/JettyServerThreadPoolMetrics.java

```diff
@@ -38,52 +44,81 @@
+
+    private final String CURRENT = "jetty.threads.current";
+
+    private final String IDLE = "jetty.threads.idle";
+
+    @Nullable
+    private MeterRegistry registry;
+
     public JettyServerThreadPoolMetrics(ThreadPool threadPool, Iterable<Tag> tags) {
         this.threadPool = threadPool;
         this.tags = tags;
     }
 
     @Override
     public void bindTo(MeterRegistry registry) {
+        this.registry = registry;
         if (threadPool instanceof SizedThreadPool) {
             SizedThreadPool sizedThreadPool = (SizedThreadPool) threadPool;
-            Gauge.builder("jetty.threads.config.min", sizedThreadPool, SizedThreadPool::getMinThreads)
+            Gauge.builder(MIN, sizedThreadPool, SizedThreadPool::getMinThreads)
                 .description("The minimum number of threads in the pool")
                 .tags(tags)
                 .register(registry);
-            Gauge.builder("jetty.threads.config.max", sizedThreadPool, SizedThreadPool::getMaxThreads)
+            Gauge.builder(MAX, sizedThreadPool, SizedThreadPool::getMaxThreads)
                 .description("The maximum number of threads in the pool")
                 .tags(tags)
                 .register(registry);
             if (threadPool instanceof QueuedThreadPool) {
                 QueuedThreadPool queuedThreadPool = (QueuedThreadPool) threadPool;
-                Gauge.builder("jetty.threads.busy", queuedThreadPool, QueuedThreadPool::getBusyThreads)
+                Gauge.builder(BUSY, queuedThreadPool, QueuedThreadPool::getBusyThreads)
                     .description("The number of busy threads in the pool")
                     .tags(tags)
                     .register(registry);
-                Gauge.builder("jetty.threads.jobs", queuedThreadPool, QueuedThreadPool::getQueueSize)
+                Gauge.builder(JOBS, queuedThreadPool, QueuedThreadPool::getQueueSize)
                     .description("Number of jobs queued waiting for a thread")
                     .tags(tags)
                     .register(registry);
             }
         }
-        Gauge.builder("jetty.threads.current", threadPool, ThreadPool::getThreads)
+        Gauge.builder(CURRENT, threadPool, ThreadPool::getThreads)
             .description("The total number of threads in the pool")
             .tags(tags)
             .register(registry);
-        Gauge.builder("jetty.threads.idle", threadPool, ThreadPool::getIdleThreads)
+        Gauge.builder(IDLE, threadPool, ThreadPool::getIdleThreads)
             .description("The number of idle threads in the pool")
             .tags(tags)
             .register(registry);
     }
 
+    @Override
+    public void close() throws Exception {
+        if (registry != null) {
+            Set<Tag> tagSet = new HashSet<>();
+            tags.forEach(tagSet::add);
+            Stream.of(MIN, MAX, BUSY, JOBS, CURRENT, IDLE)
+                .flatMap(name -> registry.find(name).meters().stream())
```

Comment:
This can fail to find the meters by name depending on the meter filters configured. That's why keeping a reference to the meters registered (or their `Id`) for directly removing them is the safe way to do this. See the implementation in `KafkaMetrics` for an example.

Reply:
Ah I see what you mean now. Thanks again, I will update this

Reply:
Ok, it is updated. Please take another look. Thanks again!